src/super_gradients/training/utils/predict/prediction_results.py

```diff
@@ -504,20 +507,21 @@ def save(
 class VideoDetectionPrediction(VideoPredictions):
     """Object wrapping the list of image detection predictions as a Video.
 
-    :attr _images_prediction_lst:   List of the predictions results
+    :attr _images_prediction_gen:   List of the predictions results
     :att fps:                       Frames per second of the video
     """
 
-    _images_prediction_lst: List[ImageDetectionPrediction]
+    _images_prediction_gen: Generator
     fps: int
+    n_frames: int
 
     def draw(
         self,
         box_thickness: int = 2,
         show_confidence: bool = True,
         color_mapping: Optional[List[Tuple[int, int, int]]] = None,
         class_names: Optional[List[str]] = None,
-    ) -> List[np.ndarray]:
+    ) -> Generator:
```

Comment:
That's a breaking change because people may have a piece of code like this
``` python
video_frames_with_preds = model.predict('video.mp4').draw()
```

Maybe we should have a method `lazy_draw` (not sure about the name) that would incorporate all the drawing logic and yield frames, use it in `draw`, `show` and `save`, but with `draw` still returning a `List[np.ndarray]`, WDYT?

Reply:
The reason for the changes here is that storing the results in a list would become painful if someone would like to detect objects on a longer video. Assuming that the "list" scenario is limited, the idea is to provide a more general approach to operating the predictions, which will become public and provide users more flexibility.

I agree with you, it's definitely a breaking change. It would be great to know @BloodAxe and @shaydeci opinions regarding this.

Reply:
My 5 cents is that returning List[np.ndarray] is still a huge risk of OOM. Yes, it is a BC, but perhaps in this particular scenario I tend to introduce it. I think in most cases people still have somthing like
```
for frame in model.predict().draw():
  ... bla bla bla ..
```

Reply:
The reason for the changes here is that storing the results in a list would become painful if someone would like to detect objects on a longer video. Assuming that the "list" scenario is limited, the idea is to provide a more general approach to operating the predictions, which will become public and provide users more flexibility.

I agree with you, it's definitely a breaking change. It would be great to know @BloodAxe and @shaydeci opinions regarding this.